controllers/status.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	ConditionCompleted = "Completed"
+	ConditionDegraded  = "Degraded"
+	ConditionResyncing = "Resyncing"
+)
+
+const (
+	Success         = "Success"
+	Promoted        = "Promoted"
+	Demoted         = "Demoted"
+	FailedToPromote = "FailedToPromote"
+	FailedToDemote  = "FailedToDemote"
+	Error           = "Error"
+	Healthy         = "Healthy"
+	ResycTriggered  = "ResyncTriggered"
+	FailedToResync  = "FailedToResync"
+	NotResyncing    = "NotResyncing"
+)
+
+// sets conditions when image was promoted successfully
```

Comment:
Also, we should consider adding timestamps as well. For example a `Completed` condition can have happened earlier, and now the resource is `Degraded` this can be gleaned from status time stamps on when each condition was observed.

Reply:
`Status.Conditions[Completed][Timestamp]` is updated whenever there is change in `Status.Conditions[Completed][Status]`